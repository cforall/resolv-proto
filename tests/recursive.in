// ensures that recursive type assertions are supported

f T | g T
g S | h S
h 1

p T | q T
q S | r S
q 1
r 1

%%

f( 1 )
p( 1 )
