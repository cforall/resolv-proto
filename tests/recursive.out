[Void / (0,1,1,-1,0)] f( 1 )
[Void / (0,1,1,-1,0)] p( 1 )
